redis/src/main/scala/zio/redis/Input.scala

```diff
@@ -139,6 +139,13 @@ object Input {
       (data._1 :: data._2).foldLeft(Chunk.empty: Chunk[String])((acc, a) => acc ++ input.encode(a))
   }
 
+  final case class NonEmptyListPrecededByCount[-A](input: Input[A]) extends Input[(A, List[A])] {
+    def encode(data: (A, List[A])): Chunk[String] =
+      (data._1 :: data._2).foldLeft(Chunk.single(wrap((1 + data._2.length).toString)): Chunk[String])((acc, a) =>
```

Comment:
Let's not introduce this one. I get the idea of calculating `numkeys` automatically, but it's not aligned with the rest of operations we expose.

Reply:
@mijicd u sure ? as a library user I would expect by lib to make it automatically

Reply:
I am sure. It is not aligned with the rest of the API. While it is quite neat, I would rather be consistent, and polish this across the API in the follow-up.

Reply:
To be perfectly clear, I think the change you propose is way friendlier, and it would be cool if you can open up an issue about it. We can use that issue to summarize all similar problems and get rid of them together.

Reply:
ok so for now, user will need to add List size as a param, but we will create new issue for it

Reply:
@mijicd @anovakovic01 
maybe you will know how to fix this:

```
3 more arguments than can be applied to method apply: (a: String, b: Long)(c: String, cs: String*)(d: Option[zio.redis.Aggregate], e: Option[List[Double]])zio.ZIO[zio.redis.RedisExecutor,zio.redis.RedisError,Long] in class Arg2VarargsArg2
            card   <- zInterStore(dest, 2, (first, List(second)), None, Some(List(2, 3)))
```

this happens now when Ive removed NonEmptyListPrecededByCount

```
RedisCommand(
    "ZINTERSTORE",
    Tuple5(
      StringInput,
      LongInput,
      NonEmptyList(StringInput),
      OptionalInput(AggregateInput),
      OptionalInput(WeightsInput)
    ),
    LongOutput
  )
```
some implicts hell ;) trying to investigate it, but its quite hard

Reply:
I think its a conflict between implicit final class Arg5 and Arg2VarargsArg2

Reply:
@mijicd are you fine with removing Arg2VarargsArg2 ? if its not used and causes problems

Reply:
@maciejbak85 I think that it's not used anywhere, so feel free to remove it (with this PR every command should be covered by tests).

Reply:
ok removed then